src/syzygy/syzygy.cc

```diff
@@ -0,0 +1,1656 @@
+            init_tb(str);
+          }
+        }
+      }
+    }
+
+    if (TB_PIECES < 7) goto finished;
+
+    for (int i = 0; i < 5; i++) {
+      for (int j = i; j < 5; j++) {
+        for (int k = j; k < 5; k++) {
+          for (int l = k; l < 5; l++) {
+            for (int m = l; m < 5; m++) {
+              sprintf(str, "K%c%c%c%c%cvK", pchr(i), pchr(j), pchr(k), pchr(l),
+                      pchr(m));
+              init_tb(str);
+            }
+          }
+        }
+      }
+    }
+
+    for (int i = 0; i < 5; i++) {
+      for (int j = i; j < 5; j++) {
+        for (int k = j; k < 5; k++) {
+          for (int l = k; l < 5; l++) {
+            for (int m = 0; m < 5; m++) {
+              sprintf(str, "K%c%c%c%cvK%c", pchr(i), pchr(j), pchr(k), pchr(l),
+                      pchr(m));
+              init_tb(str);
+            }
+          }
+        }
+      }
+    }
+
+    for (int i = 0; i < 5; i++) {
+      for (int j = i; j < 5; j++) {
+        for (int k = j; k < 5; k++) {
+          for (int l = 0; l < 5; l++) {
+            for (int m = l; m < 5; m++) {
+              sprintf(str, "K%c%c%cvK%c%c", pchr(i), pchr(j), pchr(k), pchr(l),
+                      pchr(m));
+              init_tb(str);
+            }
+          }
+        }
+      }
+    }
+
+  finished:
+    std::cerr << "Found " << num_wdl_ << "WDL, " << num_dtm_ << " DTM and "
+              << num_dtz_ << " DTZ tablebase files." << std::endl;
+  }
+
+  ~SyzygyTablebaseImpl() {
+    // if pathString was set there may be entries in need of cleaning.
+    if (!paths_.empty()) {
+      for (int i = 0; i < num_piece_entries_; i++)
+        free_tb_entry(&piece_entries_[i]);
+      for (int i = 0; i < num_pawn_entries_; i++)
+        free_tb_entry(&pawn_entries_[i]);
+    }
+  }
+
+  int max_cardinality() const { return max_cardinality_; }
+
+  int probe_wdl_table(const ChessBoard& pos, int* success) {
+    return probe_table(pos, 0, success, WDL);
+  }
+
+  int probe_dtm_table(const ChessBoard& pos, int won, int* success) {
+    return probe_table(pos, won, success, DTM);
+  }
+
+  int probe_dtz_table(const ChessBoard& pos, int wdl, int* success) {
+    return probe_table(pos, wdl, success, DTZ);
+  }
+
+ private:
+  std::string name_for_tb(const char* str, const char* suffix) {
+    std::stringstream path_string_stream(paths_);
+    std::string path;
+    std::ifstream stream;
+    while (std::getline(path_string_stream, path, SEP_CHAR)) {
+      std::string fname = path + "/" + str + suffix;
+      stream.open(fname);
+      if (stream.is_open()) return fname;
+    }
+    return std::string();
+  }
+
+  bool test_tb(const char* str, const char* suffix) {
+    return !name_for_tb(str, suffix).empty();
+  }
+
+  void* map_tb(const char* name, const char* suffix, map_t* mapping) {
+    std::string fname = name_for_tb(name, suffix);
+    void* base_address;
+#ifndef _WIN32
+    stat statbuf;
```

Comment:
This line needs to be `struct stat statbuf; `to compile on the linux build.

Reply:
Yeah, overzealous cleanup of extraneous use of the word struct.  My bad, fixed.